tests/libvmi/factory.go

```diff
@@ -33,46 +33,13 @@ const (
 
 // NewFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
+// This image has tooling for the guest agent, stress, SR-IOV and more.
 func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return NewTestToolingFedora(opts...)
-}
-
-// NewTestToolingFedora instantiates a new Fedora based VMI configuration,
-// building its extra properties based on the specified With* options.
-// This image has tooling for the guest agent, stress, and more
-func NewTestToolingFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedoraTestTooling, opts...)
-}
-
-// NewSriovFedora instantiates a new Fedora based VMI configuration,
-// building its extra properties based on the specified With* options, the
-// image used include Guest Agent and some moduled needed by SRIOV.
-func NewSriovFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedoraTestTooling, opts...)
-}
-
-// NewSEVFedora instantiates a new Fedora based VMI configuration,
-// building its extra properties based on the specified With* options, the
-// image used is configured for UEFI boot and it supports AMD SEV.
-func NewSEVFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
```

Comment:
> NewSEVFedora is being used in only one place, therefore
it seems reasonable to use the options directly at the
caller.

This function will be used for attestation tests as well. Currently its a WIP PR https://github.com/kubevirt/kubevirt/pull/7197/commits/bb14be17a6deafcdac11ddc4a0c06e11f51b034c . If this is not the right place, maybe you could just move the function to `launchsecurity/sev.go` instead of removing it?

Reply:
If you'll find it really useful, could you return it back through that PR?
Moving it now is a bit odd to me without any usages.

But have you considered just reusing `vmi := libvmi.NewFedora(libvmi.WithUefi(secureBoot), libvmi.WithSEV())` whenever you need to? We do this with other tests all the time and it is more expressive as one can see directly the options used without the need to "trust" the helper factory additions.
I think that is one of the great values of the build-pattern.